src/server/db/actions/UserAction.ts

```diff
@@ -56,3 +57,77 @@ export async function verifyUser(email: string, password: string) {
     _id: nonSensitiveUser._id.toString(),
   };
 }
+
+/**
+ * Gets a user by their ID.
+ * @param {string} id ID of the user to get.
+ * @throws {UserDoesNotExistException} If unable to find user
+ */
+export async function getUser(id: string) {
+  await connectMongoDB();
+  const user = await UserModel.findById(id);
+  if (!user) {
+    throw new UserDoesNotExistException();
+  }
+  return user;
+}
+
+/**
+ * Edits a user.
+ * @param {z.infer<typeof userSchema> & { _id: string }} userInfo Info of the user to find/update.
+ * @throws {GenericUserErrorException} If changing email to an email that corresponds to a pre-existing account
+ * @throws {UserDoesNotExistException} If unable to find user
+ * 
+ */
+export async function editUser(userInfo: z.infer<typeof userSchema> & { _id: string }) {
```

Comment:
You can give `_id` a stricter type using `zod` to look out for the length of the string (24 I believe)